File: frontend/src/types/adaptation.ts
export interface ParameterSnapshot {
  id: number;
  snapshot_type: string;
  trigger: string | null;

  // Entry parameters
  rsi_period: number;
  rsi_overbought: number;
  rsi_oversold: number;
  ema_fast: number;
  ema_slow: number;
  volume_multiplier: number;
  min_signal_strength: number;
  technical_weight: number;
  sentiment_weight: number;
  fundamental_weight: number;

  // Exit parameters
  atr_stop_multiplier: number;
  atr_target_multiplier: number;
  default_stop_loss_pct: number;
  default_profit_target_pct: number;
  max_hold_bars: number;

  full_config: Record<string, unknown> | null;
  created_at: string;
}

export interface MetaReview {
  id: number;
  review_date: string;
  regime_at_review: string | null;
  summary: string;
  recommendations: string[] | Record<string, unknown> | null;
  parameter_adjustments: string[] | Record<string, unknown> | null;
  exit_strategy_assessment: string[] | Record<string, unknown> | null;
  signals_generated: number;
  signals_correct: number;
  avg_return: number | null;
  regime_accuracy: number | null;
  created_at: string;
}

export interface DailyPerformance {
  id: number;
  perf_date: string;
  total_signals: number;
  buy_signals: number;
  sell_signals: number;
  signals_correct: number;
  signals_incorrect: number;
  win_rate: number | null;
  avg_return_pct: number | null;
  best_return_pct: number | null;
  worst_return_pct: number | null;
  total_return_pct: number | null;
  regime: string | null;
  breakdown: Record<string, unknown> | null;
  created_at: string;
}

export interface PerformanceSummary {
  days: number;
  total_signals: number;
  total_correct: number;
  overall_win_rate: number | null;
  avg_daily_return_pct: number | null;
  cumulative_return_pct: number | null;
  best_day: DailyPerformance | null;
  worst_day: DailyPerformance | null;
  daily: DailyPerformance[];
}

File: backend/app/schemas/adaptation.py
from datetime import date, datetime

from pydantic import BaseModel


class ParameterSnapshotResponse(BaseModel):
    id: int
    snapshot_type: str
    trigger: str | None = None

    # Entry parameters
    rsi_period: int
    rsi_overbought: int
    rsi_oversold: int
    ema_fast: int
    ema_slow: int
    volume_multiplier: float
    min_signal_strength: float
    technical_weight: float
    sentiment_weight: float
    fundamental_weight: float

    # Exit parameters
    atr_stop_multiplier: float
    atr_target_multiplier: float
    default_stop_loss_pct: float
    default_profit_target_pct: float
    max_hold_bars: int

    full_config: dict | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class MetaReviewResponse(BaseModel):
    id: int
    review_date: date
    regime_at_review: str | None = None
    summary: str
    recommendations: list | dict | None = None
    parameter_adjustments: list | dict | None = None
    exit_strategy_assessment: list | dict | None = None
    signals_generated: int
    signals_correct: int
    avg_return: float | None = None
    regime_accuracy: float | None = None
    created_at: datetime

    model_config = {"from_attributes": True}
